src/name.rs

```diff
@@ -43,6 +43,9 @@ impl<'a> Name<'a> {
                 return Ok(Name { labels: &data[..pos+2], original: original });
             } else if byte & 0b1100_0000 == 0 {
                 let end = pos + byte as usize + 1;
+                if data.len() < end {
```

Comment:
I don't think so since end is the ending index of the slice, it can be the same as the length of the array. If we were indexing for a specific value, then it should be less or equals to.
I think it's the same here
https://github.com/rigdent/dns-parser/blob/ec2c2f87043938adf56eb28a43040afa8823fa57/src/name.rs#L33